### PR TITLE
added types to @redwood/web, @redwood/forms

### PR DESCRIPTION
**Pull request by Burnsy**
Hey all; 

# What it does

- Adds types for redwood/web packages;
- some types have been left to any as I don't fully understand how to type them yet.
- I was also going to type forms but noticed that has moved into its own package
- changes the default types file away from global.web-auto-imports.ts (i belive the types were not even working from that file as they were not pulling through in my VS Code.) 
- left the global.web-auto-imports.ts as I don't know if it is used for anything else

### CODE REVIEW

**Comment by peterp** [comment]
Yay! Thanks @Burnsy!

**Comment by Burnsy** [comment]
I noticed I made a mistake and removed the global imports for the types, I have put them back in and it all seems to work on my typescript redwood project so yeah! not perfect but works and covers most of the types I believe. 

I believe the tasks are falling as it can't find the testing types.
``##[error]../web/src/index.d.ts(8,8): error TS7016: Could not find a declaration file for module '@redwoodjs/testing'. '/home/runner/work/redwood/redwood/packages/testing/dist/index.js' implicitly has an 'any' type.``

**Comment by Burnsy** [comment]
@peterp it could be that inside the testing package ([Line 9](https://github.com/redwoodjs/redwood/blob/main/packages/testing/package.json#L9)) it says there is a types file in the dist folder, but then when built there is no typed file so when web types look for the file it cant find the types.

**Comment by Burnsy** [comment]
I have also added types for `@redwood/testing` and `@redwood/forms` currently the all the types for the forms are unknown until I type them out. The reason they are unknown is so that the tsc imports will still find the types but then it will act as they are untyped. 

TLDR; unknown types stop the error of unfound types. until proper typing is added.

**Comment by peterp** [comment]
@Burnsy The fact that the types aren't built is something that I did; which needs to undone. You need to run `yarn lerna build:types`

I'll fix this though and ping you when I'm done.

**Comment by Burnsy** [comment]
I thought the testing library was meant to declare it types automatically (as it is typescript) 

I won't get time to progress on it today but I will wait for your ping :)

**Comment by peterp** [comment]
I've just merged #938 that should help aid in the completion of this PR.

**Comment by peterp** [comment]
Well... either aid or make it completely difficult 🤷‍♂️ to complete. My hope is that it'll make things clearer :) Let me know!

File: packages/testing/src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  setupRequestHandlers as _setupRequestHandlers,
+} from './mockRequests'
+import { MockProviders as _MockProviders } from './MockProviders'
+declare module '@redwoodjs/testing' {
+  const mockGraphQLMutation: typeof _mockGraphQLMutation
+  const mockGraphQLQuery: typeof _mockGraphQLQuery
+  const MockProviders: typeof _MockProviders
+  const startMSW: typeof _startMSW
+  const setupRequestHandlers: typeof _setupRequestHandlers
+}
```
Comment (peterp) [inline packages/testing/src/index.d.ts:16]: Hmmm... We actually do have types here. I think this file should removed.

File: packages/testing/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "dist"
   ],
   "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "types": "src/index.d.ts",
```
Comment (peterp) [inline packages/testing/package.json:9]:
```suggestion
  "types": "dist/index.d.ts",
```

Comment (Burnsy) [reply packages/testing/package.json]: So when we remove the Types in the package.json, it then allows the compiler to build its own types instead of having them types be defined by a single file. 

So if we remove the type file and the line in the package.json. ergo it will make its own types for the other files to read!
